### PR TITLE
sink/flowcontrol(ticdc): make tests run and refactor code

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/5138

### What is changed and how it works?

- make tests run and using testify
- refactor code, split file, rename fileds

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Side effects
None
Related changes
None

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/sink/flowcontrol/flow_control_test.go [hunk 1]
```diff
@@ -21,11 +21,16 @@ import (
...
 	"time"
 
+	"github.com/stretchr/testify/require"
+
 	"github.com/pingcap/check"
-	"github.com/pingcap/tiflow/pkg/util/testleak"
 	"golang.org/x/sync/errgroup"
 )
 
+func Test(t *testing.T) {
```
Comment (CharlesCheung96) [inline cdc/sink/flowcontrol/flow_control_test.go:30]: Should this statement be removed when using the testify framework?

Comment (hi-rustin) [reply cdc/sink/flowcontrol/flow_control_test.go]: Oops! Removed.

File: cdc/sink/flowcontrol/flow_control_test.go [hunk 2]
```diff
@@ -44,10 +39,8 @@ func (c *mockCallBacker) cb() error {
 	return c.injectedErr
 }
 
-func (s *flowControlSuite) TestMemoryQuotaBasic(c *check.C) {
-	defer testleak.AfterTest(c)()
-
-	controller := NewTableMemoryQuota(1024)
+func TestMemoryQuotaBasic(t *testing.T) {
+	controller := newTableMemoryQuota(1024)
```
Comment (CharlesCheung96) [inline cdc/sink/flowcontrol/flow_control_test.go:43]: Maybe we should run the tests in parallel.

Comment (hi-rustin) [reply cdc/sink/flowcontrol/flow_control_test.go:45]: Addressed. Thanks!

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**